CountWords/CountWords.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"unicode"
+)
+
+type node struct {
+	left, right *node
+	word        []byte
+	repeates    int
+}
+
+func less(a, b []byte) bool {
+	var invert bool
+	if len(a) > len(b) {
+		a, b = b, a
+		invert = true
+	}
+	for i := range a {
+		if a[i] < b[i] {
+			return !invert
+		}
+		if a[i] > b[i] {
+			return invert
+		}
+	}
+	return true
+}
+
+func insert(t *node, v []byte) *node {
+	if t == nil {
+		t = &node{word: v, repeates: 1}
+		return t
+	}
+	if reflect.DeepEqual(t.word, v) {
+		t.repeates++
+		return t
+	}
+	if less(v, t.word) {
+		t.left = insert(t.left, v)
+		return t
+	}
+	t.right = insert(t.right, v)
+	return t
+}
+
+func main() {
```

Comment:
Could you please add tests to the program?

The tests should not read/write any files. You will need to restructure your program.